internal/core/cloud/aws.go

```diff
@@ -151,11 +173,45 @@ func requestMetadata(ctx context.Context, client HTTPClient, url string) (interf
 		return nil, err
 	}
 
+	if resp.StatusCode == http.StatusUnauthorized || resp.StatusCode == http.StatusForbidden {
+		return nil, errors.Errorf("failed to fetch AWS metadata: %s", resp.Status)
+	}
+
 	// The metadata endpoint may return json elements
 	if json.Valid(body) {
-		var jsonData interface{}
+		var jsonData any
 		err := json.Unmarshal(body, &jsonData)
 		return jsonData, err
 	}
 	return string(body), nil
 }
+
+func requestMetadataToken(ctx context.Context, client HTTPClient) (string, error) {
+	log.Debug("Fetching IMDS token...")
+
+	url := fmt.Sprintf("%sapi/token", awsMetadataURL)
```

Comment:
nitpick: we could most probably use [path.Join](https://pkg.go.dev/path#Join). I know that we use `Sprintf` in many places though :sweat_smile:

Reply:
Tried it, however besides looking more related to filesystem paths, it breaks with the `http://...`

```go
url := path.Join(awsMetadataURL, "api/token")
```

returns `"http:/169.254.169.254/latest/api/token"`
Note that there's one missing `/` after `http:`